profiling/paper_timing_tables/timing_comparison.py

```diff
@@ -0,0 +1,358 @@
+    )
+
+    # copy TOAs to create 2x the number of TOAs
+    if "J1910+1256_NANOGrav_12yv4_10k.tim" in timfiles2:
+        subprocess.call(
+            "cat J1910+1256_NANOGrav_12yv4.tim > J1910+1256_NANOGrav_12yv4_10k.tim",
+            shell=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        subprocess.call(
+            "sed -n '6,6761p' J1910+1256_NANOGrav_12yv4.tim >> J1910+1256_NANOGrav_12yv4_10k.tim",
+            shell=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+
+    # copy TOAs to create 5x the number of TOAs
+    if "J1910+1256_NANOGrav_12yv4_25k.tim" in timfiles2:
+        subprocess.call(
+            "cat J1910+1256_NANOGrav_12yv4.tim > J1910+1256_NANOGrav_12yv4_25k.tim",
+            shell=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        for i in range(4):
+            subprocess.call(
+                "sed -n '6,6761p' J1910+1256_NANOGrav_12yv4.tim >> J1910+1256_NANOGrav_12yv4_25k.tim",
+                shell=True,
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
+
+    ptimes_nopickle2 = []
+    ptimes_pickle2 = []
+    ttimes2 = []
+    t2times2 = []
+
+    for tim in timfiles2:
+        # run PINT w/o pickling and average time over 5 runs
+        print("Running PINT w/o pickling...")
+        pintrun(
+            "J1910+1256_NANOGrav_12yv4.gls.par",
+            tim,
+            ptimes_nopickle2,
+            pickle=False,
+            fitter="gls",
+        )
+
+        # run PINT with pickling and average time over 5 runs
+        print("Running PINT w/ pickling...")
+        subprocess.call(
+            "pintempo --usepickle J1910+1256_NANOGrav_12yv4.gls.par " + tim,
+            shell=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )  # create pickle file
+        pintrun(
+            "J1910+1256_NANOGrav_12yv4.gls.par",
+            tim,
+            ptimes_pickle2,
+            pickle=True,
+            fitter="gls",
+        )
+
+        print("running TEMPO...")
+        temporun("J1910+1256_NANOGrav_12yv4.gls.par", tim, ttimes2, fitter="gls")
+
+        print("running Tempo2...")
+        tempo2run("J1910+1256_NANOGrav_12yv4.gls.par", tim, t2times2)
+
+    # create table 9 in PINT paper
+    complex_comparison = Table(
+        (ntoas_complex, ttimes2, t2times2, ptimes_nopickle2, ptimes_pickle2),
+        names=(
+            "Number of TOAs",
+            "TEMPO (sec)",
+            "Tempo2 (sec)",
+            "PINT - No Pickle (sec)",
+            "PINT - Pickle (sec)",
+        ),
+    )
+    ascii.write(
+        complex_comparison,
+        "complex_tables.pdf",
+        Writer=ascii.Latex,
+        latexdict={"tabletype": "table*"},
+        overwrite=True,
+    )
+
+    # remove added params from par file for future use
```

Comment:
This will not get run if an exception - including `KeyboardInterrupt` - arises during execution.

I recommend using `with TemporaryDirectory as d:` and creating and using all files in that directory - this will be deleted when you exit the `with`  block.